tools/run_tests/helper_scripts/clean_xds_interop_resources.py

```diff
@@ -0,0 +1,168 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import json
+import sys
+import re
+import os
+import functools
+import subprocess
+import datetime
+from dataclasses import dataclass
+from typing import List, Any
+
+# Parses commandline arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('--dry_run', action='store_true')
+args = parser.parse_args()
+
+# Type alias
+Json = Any
+
+# Configures this script
+KEEP_PERIOD = datetime.timedelta(hours=24)
+GCLOUD = os.environ.get('GCLOUD', 'gcloud')
+GCLOUD_CMD_TIMEOUT_S = datetime.timedelta(seconds=120).total_seconds()
+ZONE = 'us-central1-a'
+SECONDARY_ZONE = 'us-west1-b'
+PROJECT = 'grpc-testing'
+
+
+@functools.lru_cache()
+def get_expire_timestamp():
+    return (datetime.datetime.now() - KEEP_PERIOD).isoformat()
+
+
+def exec_gcloud(*cmds: List[str]) -> Json:
+    cmds = [GCLOUD, '--project', PROJECT] + list(cmds)
+    if 'list' in cmds:
+        # Add arguments to shape the list output
+        cmds.extend([
+            '--format', 'json', '--filter',
+            f'creationTimestamp<={get_expire_timestamp()}'
+        ])
+    if args.dry_run and 'delete' in cmds:
+        # Skip deletion for dry-runs
+        logging.debug('> Skipped: %s', " ".join(cmds))
+        return None
+    # Executing the gcloud command
+    logging.debug('Executing: %s', " ".join(cmds))
+    proc = subprocess.Popen(cmds,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            text=True)
+    returncode = proc.wait(timeout=GCLOUD_CMD_TIMEOUT_S)
+    if returncode:
+        logging.error('> Failed to execute cmd [%s], returned %d, stderr: %s',
+                      " ".join(cmds), returncode, proc.stderr.read())
+        return None
+    return json.loads(proc.stdout.read())
+
+
+def remove_relative_resources_run_xds_tests(suffix: str):
+    """Removing GCP resources created by run_xds_tests.py."""
+    logging.info('Removing run_xds_tests.py resources with suffix [%s]', suffix)
+    exec_gcloud('compute', 'forwarding-rules', 'delete',
+                f'test-forwarding-rule{suffix}', '--global')
+    exec_gcloud('compute', 'target-http-proxies', 'delete',
+                f'test-target-proxy{suffix}')
+    exec_gcloud('alpha', 'compute', 'target-grpc-proxies', 'delete',
+                f'test-target-proxy{suffix}')
+    exec_gcloud('compute', 'url-maps', 'delete', f'test-map{suffix}')
+    exec_gcloud('compute', 'backend-services', 'delete',
+                f'test-backend-service{suffix}', '--global')
+    exec_gcloud('compute', 'backend-services', 'delete',
+                f'test-backend-service-alternate{suffix}', '--global')
```

Comment:
This seems very prone to drift; it seems this should really be part of the test framework. Is there a reason we're making this a separate script and not just doing an extra cleanup at the beginning/end of the test frameworks? That's what we've done for some other similar cases in the past.

I could understand this script if we plan to trash it after a quarter or so, but it didn't sound like how this was proposed.

Reply:
Updated to use the "delete button" in our test scripts.

Agreed. The benefit of this approach is simple and intuitive, but might be out-of-date soon. I updated this script to invoke the clean up mode for our two other scripts.

There is a downside though, our test script requires many additional dependencies to start. This is a problem for manual runs, but CI won't care. (New Python packages and versions, and compiling ProtoBuf)

Reply:
Still unanswered: Is there a reason we're making this a separate script and not just doing an extra cleanup at the beginning/end of the test frameworks?